web/pgadmin/browser/static/js/keyboard.js

```diff
@@ -123,35 +143,52 @@ _.extend(pgBrowser.keyboardNavigation, {
     }
   },
   bindRightPanel: function(event, combo) {
-    let allPanels = pgAdmin.Browser.docker.findPanels();
-    let activePanel = 0;
-    let nextPanel = allPanels.length - 1;
-    let prevPanel = 1;
-    let activePanelId = 0;
-    let activePanelFlag = false;
+    const self = this;
+    let dockLayoutTabs = document.activeElement?.closest('.dock-layout')?.querySelectorAll('.dock-tab-btn');
     let shortcut_obj = this.keyboardShortcut;
-
-    _.each(pgAdmin.Browser.docker.findPanels(), (panel, index) => {
-      if (panel.isVisible() && !activePanelFlag && panel._type !== 'browser') {
-        activePanelId = index;
-        activePanelFlag = true;
+    if (document.activeElement.closest('.dock-tab-btn')) {
+      let currDockTab = document.activeElement?.closest('.dock-tab-btn');
+      if(dockLayoutTabs?.length > 1 && currDockTab) {
+        for(let i=0; i<dockLayoutTabs.length; i++) {
+          if(dockLayoutTabs[i] == currDockTab) {
+            let activeTabIdx = i;
+            self._focusTab(dockLayoutTabs, activeTabIdx, shortcut_obj, combo);
+            break;
+          }
+        }
       }
-    });
-
-    if (combo === shortcut_obj.tabbed_panel_backward) activePanel = (activePanelId > 0) ? activePanelId - 1 : prevPanel;
-    else if (combo === shortcut_obj.tabbed_panel_forward) activePanel = (activePanelId < nextPanel) ? activePanelId + 1 : nextPanel;
-
-    pgAdmin.Browser.docker.findPanels()[activePanel].focus();
-    setTimeout(() => {
-      if (document.activeElement instanceof HTMLIFrameElement) {
-        document.activeElement.blur();
+    } else if (document.activeElement.nodeName === 'IFRAME' || document.activeElement.closest('.dock-tabpane.dock-tabpane-active')?.id) {
+      let activeTabId = '';
+      if (document.activeElement.nodeName === 'IFRAME'){
+        dockLayoutTabs = document.activeElement?.closest('#root')?.querySelectorAll('.dock-tab-btn');
+        activeTabId = document.activeElement?.id;
+      } else if (document.activeElement.closest('.dock-tabpane.dock-tabpane-active')?.id){
+        activeTabId = document.activeElement.closest('.dock-tabpane.dock-tabpane-active')?.id;
+      }
+      if(dockLayoutTabs?.length > 1 && activeTabId) {
```

Comment:
In one condition we are checking directly with the node element when user clicks on tab while in the other condition we are checking with id, as the user clicked the any field within the rc-dock content holder. It would be good to make it separate.

Reply:
then where are comments? There should be some comments to explain complex logics.

Reply:
Done